iceoryx_examples/automotive_soa/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.16)
+project(example_automotive_soa)
+
+include(GNUInstallDirs)
+
+find_package(iceoryx_posh CONFIG REQUIRED)
+find_package(iceoryx_hoofs CONFIG REQUIRED)
+
+get_target_property(ICEORYX_CXX_STANDARD iceoryx_posh::iceoryx_posh CXX_STANDARD)
+include(IceoryxPlatform)
+
+option(USE_UDS "Send Events via UNIX domain sockets" OFF)
+
+add_executable(iox-cpp-automotive-skeleton  ./iox_automotive_skeleton.cpp
+                                            ./src/minimal_skeleton.cpp
+                                            ./src/runtime.cpp
+                                            ./src/owl/kom/method_server.cpp
+)
+target_link_libraries(iox-cpp-automotive-skeleton
+    iceoryx_posh::iceoryx_posh
+)
+target_compile_options(iox-cpp-automotive-skeleton PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
```

Comment:
Defining `PRIVATE` or `PUBLIC` does not make sense for executables. Could you please remove those keywords for every executable target.

For libraries it is used to add the linker flags, include dirs or options in targets which depend on them but it is impossible to depend on an executable.

Reply:
From [CMake docs](https://cmake.org/cmake/help/latest/command/target_compile_options.html):

> The INTERFACE, PUBLIC and PRIVATE keywords are required to specify the scope of the following arguments.

For the other examples we have set them to `PRIVATE`, but as you stated it doesn't really matter for exectuables.

Reply:
I know we used them also in other targets but we do not need to continue this mistake :D